Add recursive user manifests

markusheinemann opened the pull request:
see #1313 

# How to Test

1. Create the following files inside of `extra-manifests`:

#### **`kustomization.yaml.tpl`**
```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

resources:
  - subfolder
```

#### **`subfolder/kustomization.yaml.tpl`**

```yaml
apiVersion: kustomize.config.k8s.io/v1beta1
kind: Kustomization

resources:
  - namespace.yaml
```

#### **`subfolder/namespace.yaml.tpl`**

```yaml
apiVersion: v1
kind: Namespace
metadata:
  name: mynamespace
```

2. Create a cluster using `terrafom apply`

3. Check if the resource has been applied: `kubectl get ns`

--> It work's on my machine 🤣

mysticaltech reviewed (changes requested):
@markusheinemann Thanks for this. Looking good. Please if you could tweak the docs for this, especially in the examples section in the readme and other relevant places (if this is mentioned there, like kube.tf), it would be great.

markusheinemann commented:
> @markusheinemann Thanks for this. Looking good. Please if you could tweak the docs for this, especially in the examples section in the readme and other relevant places (if this is mentioned there, like kube.tf), it would be great.

I have updated the related docs and updated the examples as well. I am happy about your feedback.

markusheinemann commented:
Actual the provided solution will not work due to https://github.com/hashicorp/terraform/issues/16330

markusheinemann commented on kustomization_user.tf:
```diff
@@ -1,5 +1,5 @@
+  user_kustomization_templates = try(fileset("extra-manifests", "**/*.yaml.tpl"), toset([]))
 }
 
 resource "null_resource" "kustomization_user" {
```
Possible solution to solve the terraform file provisioner is to create the directories before copying

```
  provisioner "remote-exec" {
    inline = [
      "mkdir -p $(dirname /var/user_kustomize/${each.key})"
    ]
  }
```

mysticaltech reviewed (approved):
Awesome @markusheinemann! Please let me know when it's working.

mysticaltech commented:
@markusheinemann What about now?

markusheinemann commented:
lgtm. I think the PR is done 💯

markusheinemann commented:
lgtm. I think the PR can be merged in my eyes. Can you please rerun the workflow?

PS. Sorry for closing the PR due to an oversight
